### PR TITLE
Add proper support for loop detection.

### PR DESCRIPTION
**Pull request by davidv1992**
This also needed to refactor how we deal with reference ids a bit.

### CODE REVIEW

File: ntp-proto/src/identifiers.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        self.0 == ReferenceId::KISS_DENY
+    }
+
+    pub(crate) fn is_rate(&self) -> bool {
+        self.0 == ReferenceId::KISS_RATE
+    }
+
+    pub(crate) fn is_rstr(&self) -> bool {
+        self.0 == ReferenceId::KISS_RSTR
+    }
```
Comment (folkertdev) [inline ntp-proto/src/identifiers.rs:37]: use `Self::CONSTANT_NAME` here?

File: ntp-proto/src/identifiers.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    pub(crate) fn to_bits(self) -> [u8; 4] {
```
Comment (folkertdev) [inline ntp-proto/src/identifiers.rs:39]: I'd name this `from_bytes`

File: ntp-proto/src/identifiers.rs (hunk 1 above)
Comment (davidv1992) [reply ntp-proto/src/identifiers.rs]: done

File: ntp-proto/src/identifiers.rs (hunk 2 above)
Comment (davidv1992) [reply ntp-proto/src/identifiers.rs]: done

**Review by folkertdev — APPROVED** [review]

**Pull request merged**